### ISSUE TITLE
SimPO : Simple preference optimization with a reference-free reward

### ISSUE THREAD

**Issue by runhani** [paperswithlove/papers-we-read#56]
## Some Links

model : [https://huggingface.co/princeton-nlp/gemma-2-9b-it-SimPO](https://huggingface.co/princeton-nlp/gemma-2-9b-it-SimPO)
code : [https://github.com/princeton-nlp/SimPO](https://github.com/princeton-nlp/SimPO)
paper : [https://arxiv.org/pdf/2405.14734](https://arxiv.org/pdf/2405.14734)

## 아니 open-ko-llm-leaderboard에 무슨일?

- 오랜만에 들어가서 확인한 [open-ko-llm-leaderboard](https://huggingface.co/spaces/upstage/open-ko-llm-leaderboard)
- 한국어로 finetuning한 모델도 아닌데 성능이 좋은 모델이 있다고? 
- 이제 DPO에서 SimPO의 시대로 전환 되는 것일까?

## 핵심 idea

![image](https://github.com/user-attachments/assets/f6a361b9-5d21-4dd2-82bc-fca6852207ed)

1.  reward를 평균으로 한다고?
- length normalized reward = average(log prob of a sequence)
2. margin을 준다고?
- wining 하고 losing하는 응답이 특정 margin (r)을 넘는 경우에만 loss값에 반영 

## 학습은 어떻게 했을까?

1. [ultrachat_200k](https://huggingface.co/datasets/HuggingFaceH4/ultrachat_200k) 아주 heavy하게 filter된 고품질 data로 base모델을 SFT모델로 바꾸고
2. [ultrafeedback](https://huggingface.co/datasets/HuggingFaceH4/ultrafeedback_binarized) data로 SimPO학습

## 아래는 그동안 나온 Preference Optimization 관련 목적함수들

![image](https://github.com/user-attachments/assets/5bbe1ee7-ca14-48aa-8027-f7a9c872a714)

## 결국... 

- resource가 없는 경우에는 이렇게 살아남아야 한다...
- Fine-tuning the [google/gemma-2-9b-it](https://huggingface.co/google/gemma-2-9b-it) on [princeton-nlp/gemma2-ultrafeedback-armorm](https://huggingface.co/datasets/princeton-nlp/gemma2-ultrafeedback-armorm) takes 8xH100 gpus 로 1시간 40분 학습
- 아무리 gemma-2-9b-it가 좋은 모델이라고 해도? 한국어도 없는 data로 finetuning했는데 성능이 높다니...
- 기술이 발전도 너무 빠르고 scale경쟁도 숨막히는 이 시대에 이런 논문 정말 단비 같다...
- 그렇다면... MLLM에도 당연히 적용가능하지 않을까?
